ge/org.osate.ge.swt/src/org/osate/ge/swt/classifiers/PrototypeBindingsField.java

```diff
@@ -0,0 +1,120 @@
+ * files ("Third Party Software"). The Third Party Software that is used by this program is dependent upon your system
+ * configuration. By using this program, You agree to comply with any and all relevant Third Party Software terms and
+ * conditions contained in any such Third Party Software or separate license file distributed with such Third Party
+ * Software. The parties who own the Third Party Software ("Third Party Licensors") are intended third party benefici-
+ * aries to this license with respect to the terms applicable to their Third Party Software. Third Party Software li-
+ * censes only apply to the Third Party Software and not any other portion of this program or this program as a whole.
+ */
+package org.osate.ge.swt.classifiers;
+
+import java.util.Objects;
+import java.util.function.Consumer;
+
+import org.eclipse.jface.layout.GridDataFactory;
+import org.eclipse.jface.layout.GridLayoutFactory;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.osate.ge.swt.ChangeEvent;
+import org.osate.ge.swt.internal.InternalUtil;
+import org.osate.ge.swt.util.SwtTestUtil;
+
+/**
+ * A component which allows selecting view and editing bindings for a node provided by a {@link PrototypeBindingsModel}.
+ *
+ * @param <N> is the type of the node being edited.
+ * @param <D> is the type of the direction options.
+ * @param <T> is the type of the type options.
+ * @param <C> is the type of the classifiers.
+ */
+public final class PrototypeBindingsField<N, D, T, C> extends Composite {
+	private static final String WIDGET_ID_PREFIX = "org.osate.ge.swt.classifiers.prototypeBindingsField.";
+	public static final String WIDGET_ID_SELECTED_LABEL = WIDGET_ID_PREFIX + "selectedLabel";
+	public static final String WIDGET_ID_EDIT_BUTTON = WIDGET_ID_PREFIX + "editButton";
+
+	private final PrototypeBindingsModel<N, D, T, C> model;
+	private N node;
+	private final Label selectedLbl;
+	private final Button editBtn;
+	private final Consumer<ChangeEvent> changeListener = e -> refresh();
+
+	/**
+	 * Create a new instance.
+	 * @param parent is the container for the new component.
+	 * @param model provides the information for the component.
+	 * @param initNode is the node from the model for which the component displays and sets values.
+	 */
+	public PrototypeBindingsField(final Composite parent, final PrototypeBindingsModel<N, D, T, C> model,
+			final N initNode) {
+		super(parent, SWT.NONE);
+		this.model = Objects.requireNonNull(model, "model must not be null");
+		this.node = initNode;
+		InternalUtil.setColorsToMatchParent(this);
+
+		this.setLayout(GridLayoutFactory.fillDefaults().numColumns(2).create());
+
+		this.selectedLbl = new Label(this, SWT.BORDER);
```

Comment:
selectedLbl seems like a bad name here, probably should be bindingsLbl or something in my opinion.  I believe this was a copy and pasted name from ClassifierWithBindingsField, in which case selectedLbl is actually the selected classifier and makes sense.

If selectedLbl name is intended you can disregard.

Also when the text in the selectedLbl runs off the label, the text is cut off and cannot be scrolled.  This isn't a big deal, users can click the Edit... to see the full list, but I do think it would be nicer to use StyledText (with editable set to false and caret set to null) so users can view the cut off text if needed.

Reply:
Renamed in new push. Using CLabel now. User will need to select edit to view remainder.